cluster-autoscaler/utils/kubernetes/factory.go

```diff
@@ -57,5 +59,26 @@ func getCorrelationOptions() kube_record.CorrelatorOptions {
 		QPS:          defaultQPS,
 		BurstSize:    defaultBurstSize,
 		LRUCacheSize: defaultLRUCache,
+		SpamKeyFunc:  getCustomSpamKeyFunc(),
+	}
+}
+
+func getCustomSpamKeyFunc() kube_record.EventSpamKeyFunc {
```

Comment:
Can you add a comment describing what this does and why? I know it, but without having all the context beforehand I think this would be very hard to understand.

Reply:
Added a comment with description